### PR TITLE
Fix stuck heads

### PR DESCRIPTION
**Pull request by MaxFomenkov**
**Problem:** if we have a merged head with priority from choice and head having most priority stuck - other heads hove no ability to propagate new heads and the whole multistream head stuck. 
**Fix:** if multistream becomes unavailable we remove it from the multistream and returns back once it ok 
Additionally added metrics for 

- heads have no updates over 60sec `stuck_head`
- current head height for every head `current_head`

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/startup/ConfiguredUpstreams.kt [hunk 1]
```diff
@@ -110,8 +113,20 @@ open class ConfiguredUpstreams(
...
                 upstream?.let {
-                    val event = UpstreamChangeEvent(chain, upstream, UpstreamChangeEvent.ChangeType.ADDED)
-                    eventPublisher.publishEvent(event)
+                    Flux.concat(Mono.just(UpstreamChangeEvent.ChangeType.ADDED), upstream.observeStatus())
+                        .distinctUntilChanged()
+                        .subscribeOn(Schedulers.boundedElastic())
+                        .subscribe { status ->
+                            when (status) {
+                                UpstreamAvailability.UNAVAILABLE -> UpstreamChangeEvent.ChangeType.REMOVED
+                                else -> UpstreamChangeEvent.ChangeType.REVALIDATED
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/startup/ConfiguredUpstreams.kt:122]: I thought nobody used this status

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt [hunk 2]
```diff
@@ -119,19 +121,31 @@ abstract class Multistream(
...
+            }
+        }
 
-    fun removeUpstream(id: String) {
-        if (upstreams.removeIf { it.getId() == id }) {
-            onUpstreamsUpdated()
-            setHead(updateHead())
+    fun removeUpstream(id: String): Boolean =
+        upstreams.removeIf { up ->
+            up.takeIf { up.getId() == id }
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt:139]: really hard to comprehend :(

**Review by a10zn8 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/emeraldpay/dshackle/startup/ConfiguredUpstreams.kt (hunk 1 above)
Comment (MaxFomenkov) [reply src/main/kotlin/io/emeraldpay/dshackle/startup/ConfiguredUpstreams.kt:122]: it has same behaviour as a ADDED

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt (hunk 2 above)
Comment (MaxFomenkov) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt]: switched to a bit classic way

**Review by a10zn8 — APPROVED** [review]

**Pull request merged**